### PR TITLE
window abstraction to support the creation of multiple windows.

### LINKED ISSUE #12: Create window class to abstract window creation code

### PR DESCRIPTION
**Pull request by farzonl**
PR Resolves #12

### CODE REVIEW

**Comment by farzonl** [comment]
@DanK2029  For this PR since I don't know a great way to automate UI testing will require a manual run. You will want to confirm that when you press a or b it only registers in the active window that registered that specific callback. In this case when test window is active you should see a print out only when you press a. When test2 window is active you should only see a print out when you press b.

**Comment by farzonl** [comment]
More changes will be coming to support PixelBuffers. The plan is to do it in a way that supports running draw calls using a lambda chain.

example of what it might look like:
https://github.com/farzonl/2dEngine/blob/ece714dc6897ac2eb62e6fb0da0286225f061508/2DEngine/WindowMgr.cpp#L88-L95

File: src/window/glfwWindow.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  std::unordered_map<GLFWwindow *, GWindowMgr *> windowMap;
+
+public:
+  void insertNewWindow(GWindowMgr *window);
+  GWindowMgr *getWindowMgr(GLFWwindow *window);
+  static void keyCallback(GLFWwindow *window, int key, int scancode, int action,
+                          int mods);
+};
+
+typedef SingletonBase<KeyCallbackStruct> Singleton;
```
Comment (farzonl) [inline src/window/glfwWindow.cpp:22]: Ended up requiring a singleton here b\c glfw is a c api and c doesn't allow you to pass lambdas as a function ptr, which means I couldn't do a lamdba capture of the KeyCallBacks vector.

**Comment by DanK2029** [comment]
Can confirm that the callbacks work on their respective window.

**Review by DanK2029 — APPROVED** [review]
Looks good to me

**Pull request merged**